From e5ddbb1fb9f872b333268d6b7cb8fab238f9b90e Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Tue, 17 Feb 2015 18:55:51 -0200
Subject: [PATCH] QMX6: U-BOOT: Enable Q7 GPIO pins use in environment scripts
Organization: O.S. Systems Software LTDA.

The Q7 GPIO pins are now accessible from the environment script. Those
are done as following

  Q7 pin  |  GPIO number
 ------------------------
  GPIO 0  |  130
  GPIO 1  |  86
  GPIO 2  |  122
  GPIO 3  |  123
  GPI     |  111
  GPO 0   |  110
  GPO 1   |  32
 ------------------------

So to enable the GPIO 0, user needs to use:

 $: gpio set 130

while to disable it, use:

 $: gpio clear 130

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
---
 board/freescale/cgt_qmx6/cgt_qmx6.c | 21 +++++++++++++++++++++
 include/configs/cgt_qmx6_common.h   |  3 +++
 2 files changed, 24 insertions(+)

diff --git a/board/freescale/cgt_qmx6/cgt_qmx6.c b/board/freescale/cgt_qmx6/cgt_qmx6.c
index 8469fc2..46bd0f7 100644
--- a/board/freescale/cgt_qmx6/cgt_qmx6.c
+++ b/board/freescale/cgt_qmx6/cgt_qmx6.c
@@ -972,6 +972,23 @@ u32 get_board_rev(void)
 }
 #endif
 
+#ifdef CONFIG_CMD_GPIO
+static void gpio_pins(void)
+{
+	static iomux_v3_cfg_t const gpio_pads[] = {
+		MX6_PAD_EIM_A25__GPIO_5_2     | MUX_PAD_CTRL(NO_PAD_CTRL), /* Q7, GPIO 0 (#130) */
+		MX6_PAD_EIM_D22__GPIO_3_22    | MUX_PAD_CTRL(NO_PAD_CTRL), /* Q7, GPIO 1 (#86) */
+		MX6_PAD_DISP0_DAT5__GPIO_4_26 | MUX_PAD_CTRL(NO_PAD_CTRL), /* Q7, GPIO 2 (#122) */
+		MX6_PAD_DISP0_DAT6__GPIO_4_27 | MUX_PAD_CTRL(NO_PAD_CTRL), /* Q7, GPIO 3 (#123) */
+		MX6_PAD_KEY_ROW4__GPIO_4_15   | MUX_PAD_CTRL(NO_PAD_CTRL), /* Q7, GPI    (#111) */
+		MX6_PAD_KEY_COL4__GPIO_4_14   | MUX_PAD_CTRL(NO_PAD_CTRL), /* Q7, GPO 0  (#110) */
+		MX6_PAD_GPIO_0__GPIO_1_0      | MUX_PAD_CTRL(NO_PAD_CTRL), /* Q7, GPO 1  (#32) */
+	};
+
+	imx_iomux_v3_setup_multiple_pads(gpio_pads, ARRAY_SIZE(gpio_pads));
+}
+#endif
+
 int board_late_init(void)
 {
 	int ret = 0;
@@ -986,6 +1003,10 @@ int board_late_init(void)
 		return -1;
 #endif
 
+#ifdef CONFIG_CMD_GPIO
+	gpio_pins();
+#endif
+
 #ifdef CONFIG_ENV_IS_IN_MMC
 	board_late_mmc_env_init();
 #endif
diff --git a/include/configs/cgt_qmx6_common.h b/include/configs/cgt_qmx6_common.h
index 31a7b7e..aa7a654 100644
--- a/include/configs/cgt_qmx6_common.h
+++ b/include/configs/cgt_qmx6_common.h
@@ -84,6 +84,9 @@
 #define CONFIG_PHYLIB
 #define CONFIG_PHY_ATHEROS
 
+/* GPIO support for environment scripts */
+#define CONFIG_CMD_GPIO
+
 /* allow to overwrite serial and ethaddr */
 #define CONFIG_ENV_OVERWRITE
 #define CONFIG_CONS_INDEX              1
-- 
2.1.4

